### ISSUE TITLE
Support generic StringToVar maps

### ISSUE THREAD

**Issue by SOF3** [spf13/pflag#375]
Support any `Value` implementation in `map[string]V`.

If the `V` implementation expects a comma, the user should be aware that this does not work correctly.

**Comment by fredbi** [comment]
@SOF3 at the moment, this is not available. Hopefully, this repo will adopt generics some day.
Meanwhile, you may take a look at a generic implementation on top of pflag, that does just that:
https://github.com/fredbi/gflag

**Comment by tomasaschan** [comment]
This issue has not been updated in a long time. Is it still relevant?

(If I receive no answer before the end of the summer, I'll close it. Feel free to re-open if you come here after that happens and it is relevant to you - but if you do, please also comment and specify what outcome you're interested in!)

**Comment by SOF3** [comment]
Yes, unless you classify any form of generics as a non-goal for this project (understandable since minimum supported go version is 1.12)

**Comment by tomasaschan** [comment]
I think generics is definitely in scope - there's a lot of code duplication in the code base that could probably be greatly reduced using generics - but bumping the supported Go version to (at least) 1.18 is a breaking change, so it would have to be in 2.0. I have started thinking - at the back of my head, nothing concrete yet - about what the path to 2.0 looks like, so that doesn't mean this won't happen, but it does mean it's a fairly long-term thing.
